### PR TITLE
Basic select

### PR DESCRIPTION
**Pull request by nastiatikk**
Status: **Opened for visibility**

Reviewers: @tedtate @kpeatt @jeffkamo @ry5n @avelinet @cole-sanderson @mlworks @yourpalsonja 

Simple select component was created with a new files structure. 
#### To Do:
- [ ] Add `:focus` and `:active` styles
- [ ] This component originally was dependent on [PR-67](https://github.com/mobify/vellum/pull/67) Vellum `$forms__variable-name` changes that were cancelled. Need to replace those variables
- [ ] Come with better form variables across Vellum and Select component

### CODE REVIEW

File: src/_style.scss [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        bottom: 0;
+
+        margin: auto; // 2
+    }
+
+
+    // Select
+    // ---
+    //
+    // 1. Relative to the parent .c-select
```
Comment (kpeatt) [inline src/_style.scss:87]: Is this comment really necessary?

**Comment by kpeatt** [comment]
Should we add some accessibility stuff in here? Focus styles? Active styles?

**Comment by ry5n** [comment]
@kpeatt Both; we’ll need some lightweight JS for focus at least.

File: src/_style.scss (hunk 1 above)
Comment (nastiatikk) [inline src/_style.scss:87]: Not at all. Just `relative` is necessary on the parent.

**Comment by kpeatt** [comment]
@ry5n I wonder... what if we just gave the select a darker border on focus?

**Comment by ry5n** [comment]
I say we leave the focus styles to the browser defaults. I just realized that the select isn’t completely invisible here, so the focus outline on the native select will still show up. Just checked it out and it does work. We just need active styles on `.c-select`.

File: src/_style.scss (hunk 1 above)
Comment (jeffkamo) [inline src/_style.scss:87]: I like seeing comments like this as a general practice. Helps ensure there is no ambiguity and clarifies the purpose of stand alone properties when related properties are far out of sight.

**Comment by jeffkamo** [comment]
Would that work? Wouldn't technically `.c-select select` be the active element?

**Comment by ry5n** [comment]
@jeffkamo Yes, the `<select>` is still what you’re interacting with. So for focus, active, etc, you’d need to style the select, not the wrapper, which isn’t very intuitive. We could choose to add JS to reflect the state of the native component onto the container.

**Comment by nastiatikk** [comment]
With current styles `select` has `border: 0` which makes styling `&:active` and `&:focus` borders impossible. I see the only way to style them with `box-shadow`, but I have a feeling that it will have some issues with androids.

**Comment by jeffkamo** [comment]
you could also use `outline` as an alternative to border, depending on what you want to do.

**Comment by nastiatikk** [comment]
Outline property doesn't inherit `border-radius` property unfortunately

**Comment by jeffkamo** [comment]
Hmm, good point. Then ya box shadow is probably fine assuming there are no performance concerns we should keep in mind. Maybe minimize any use of excess blur?

**Comment by nastiatikk** [comment]
For some reason if I remove blur colour becomes pale:

`box-shadow: 0 0 0 1px #16a085;`
screenshot 2015-04-08 14 57 47

`box-shadow: 0 0 1px #16a085;`
screenshot 2015-04-08 14 57 32

**Comment by jeffkamo** [comment]
That's because your expand (??) parameter is becoming the blur property. So not the shadow starts behind the element and blurs out by 1px, creating the illusion that you have a 1px border that's paler than your hex color.

**Comment by jeffkamo** [comment]
Closing this PR. Any remaining work is being completed in #4

**Pull request closed without merging**